### DISCUSSION TITLE
[RFC] 077 - 兼容 Postgres/Pglite 的数据导入导出

### DISCUSSION THREAD

**Discussion by arvinxx** [lobehub/lobehub#5147] [RFCs]
## 背景

在 #4868 中我们完成了 client db 切换到 pglite，但目前反馈了一个比较大的问题： https://github.com/lobehub/lobe-chat/issues/5144 。

这个问题的根因出在现在的导出实现是为 dexie 时期的db制作的，而不是针对 postgres 这样标准的关系型数据库。

因此，我们需要为新时期的 pglite和 postgres 提供一个新的数据导入导出实现。

### 数据导出

导出的核心逻辑相对来说应该比较简单，将当前 db 中的关键表，按用户维度做一次导出，不进行任何修改，然后在导出 json 中包含所有表记录的结构即可。

注意事项:
1. 需要在导出字段中加一个标记，区分这是 postgres db 数据还是 dexie db 数据
2. 不使用同一个字段记录 state ，尽量把两个时期的导出配置做出明显区分度，方便未来检查
3. 要考虑下 db 版本的 migration 该怎么做
4. 不允许 postgres 的数据再导入回 dexiedb，保证单向数据迁移，避免未来的问题

完成该功能后，我们应该就能实现 pglite 数据导入 serverDB 版，serverDB 也可以导出数据导入回 pglite。

- https://github.com/lobehub/lobe-chat/issues/4212
- #3193

最近在具体实现过程中发现一个问题就是目前的 database 有不少表是不支持 userId 的，这对于导出数据有比较大的挑战，因此先准备做一轮 database sql 的迁移，进而实现更加完善的表结构逻辑。

## 进展

- 所有表均添加 user Id https://github.com/lobehub/lobe-chat/pull/7022
- 添加导入用的 client id https://github.com/lobehub/lobe-chat/pull/7089
- #5581

**Comment by BrandonStudio** [comment]
所以最终设计是什么样的？沿用冗余client_id还是改主键？

**Reply by arvinxx** [reply]
冗余 client_id
